aten/src/ATen/core/Device.cpp

```diff
@@ -46,29 +57,21 @@ std::pair<Device::Type, size_t> parse_type(const std::string& device_string) {
 //   index_ = std::stoi(match[3].str());
 // }
 Device::Device(const std::string& device_string) : Device(Type::CPU) {
-  AT_CHECK(!device_string.empty(), "Device string must not be empty");
+  std::istringstream ss(device_string);
```

Comment:
Errr.... I had no problems with the old code, but if we're going to optimize, isn't stringstream/getline one of those C++ standard library functions that is famously slow?

Reply:
Yeah, that will be super slow. It will copy `device_string`, and is really not needed in our simple use case.